readthedocs/core/static-src/core/js/doc-embed/search.js

```diff
@@ -51,61 +51,43 @@ function attach_elastic_search_query_sphinx(data) {
 
         search_url.href = data.proxied_api_host + '/api/v2/docsearch/';
         search_url.search = '?q=' + $.urlencode(query) + '&project=' + project +
-                            '&version=' + version + '&language=' + language;
+                            '&version=' + version + '&language=' + language + '&new-api=true';
 
         search_def
             .then(function (data) {
-                var hit_list = data.results || [];
+                var results = data.results || [];
 
-                if (hit_list.length) {
-                    for (var i = 0; i < hit_list.length; i += 1) {
-                        var doc = hit_list[i];
-                        var highlight = doc.highlight;
-                        var inner_hits = doc.inner_hits || [];
+                if (results.length) {
+                    for (var i = 0; i < results.length; i += 1) {
+                        var result = results[i];
+                        var blocks = result.blocks;
                         var list_item = $('<li style="display: none;"></li>');
 
-                        var title = doc.title;
-                        // if highlighted title is present,
-                        // use that.
-                        if (highlight) {
-                            if (highlight.title) {
-                                title = xss(highlight.title[0]);
-                            }
+                        var title = result.title;
+                        // if highlighted title is present, use that.
+                        if (result.highlights.title.length) {
+                            title = xss(result.highlights.title[0]);
                         }
 
-                        var link = doc.link + "?highlight=" + $.urlencode(query);
+                        var link = result.link + "?highlight=" + $.urlencode(query);
 
                         var item = $('<a>', {'href': link});
 
                         item.html(title);
                         item.find('span').addClass('highlighted');
                         list_item.append(item);
 
-                        // If the document is from subproject, add extra information
-                        if (doc.project !== project) {
-                            var text = " (from project " + doc.project + ")";
+                        // If the document is from a subproject, add extra information
+                        if (result.project !== project) {
+                            var text = " (from project " + result.project + ")";
                             var extra = $('<span>', {'text': text});
                             list_item.append(extra);
                         }
 
-                        for (var j = 0; j < inner_hits.length; j += 1) {
+                        for (var j = 0; j < blocks.length; j += 1) {
```

Comment:
I'd like to see better variables here than `j`. It's used a bunch and could be clearer.